### PR TITLE
Add screen condition to conditional card.

### PR DESCRIPTION
**Pull request by piitaya**
## Proposed change

**New condition type : screen.**

This will be used in the energy dashboard to hide the date card on desktop so we don't have to have dynamic config based on the screen size in energy strategy.
The screen condition has only one parameter `media_query`. The user can define any media query using yaml. 
In the UI, 4 screen sizes are displayed (mobile, tablet, desktop, wide). The editor will build the media query according to the user selection.

**New editor**

- Migration to `ha-form`
- A toggle icon has been added to switch between ui and yaml editor.
- A delete icon has been added to delete the condition. Before, the user had to click on the clear button in the entity picker which resulted to an error. This bug has been fixed too.
- The state input now uses state translations.
- A "add condition" button to select the condition type.

CleanShot 2023-10-03 at 11 16 39

CleanShot 2023-10-03 at 11 16 52

CleanShot 2023-10-03 at 11 17 04

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

**Comment by bramkragten** [comment]
We should make the UI editor backwards compatible, if there is an `entity` param in the condition, we should add `condition: state` to it so the UI editor will just work.

File: src/translations/en.json [hunk 1]
```diff
@@ -4755,6 +4755,8 @@
               "state_equal": "State is equal to",
               "state_not_equal": "State is not equal to",
               "current_state": "current",
+              "min_width": "Min width",
+              "max_width": "Max width",
```
Comment (bramkragten) [inline src/translations/en.json:4759]: unused atm?

Comment (piitaya) [reply src/translations/en.json:4759]: No, I'm working on the UI editor for the responsive condition. I will remove unused keys.

File: src/panels/lovelace/components/hui-conditional-base.ts [hunk 2]
```diff
@@ -47,27 +54,84 @@ export class HuiConditionalBase extends ReactiveElement {
...
+
+    if (deepEqual(mediaQueries, this._mediaQueries)) return;
+
+    this._mediaQueries = mediaQueries;
+    while (this._mediaQueriesListeners.length) {
+      this._mediaQueriesListeners.pop()!();
+    }
+    mediaQueries.forEach((query) => {
+      const listener = listenMediaQuery(query, () => {
+        this._updateVisibility();
```
Comment (bramkragten) [inline src/panels/lovelace/components/hui-conditional-base.ts:96]: Wonder if we can this more efficient, so we don't have to check the media query again in `_updateVisibility` 🤔  can get complicated if there are multiple, but maybe we should optimise for the most common case of 1 media query/condition?

Comment (bramkragten) [inline src/panels/lovelace/components/hui-conditional-base.ts:96]: Something like:
```suggestion
	if (this._config!.conditions.length === 1) {
		this._setVisible(ev.matches);
	} else {
        	this._updateVisibility();
        }
```

**Comment by piitaya** [comment]
I added screen sizes to the UI, I improved the editor and improved performance for the case with only one condition.

File: src/panels/lovelace/editor/config-elements/hui-conditional-card-editor.ts [hunk 3]
```diff
@@ -352,22 +325,13 @@ export class HuiConditionalCardEditor
         .condition {
           margin-top: 8px;
           border: 1px solid var(--divider-color);
```
Comment (bramkragten) [inline src/panels/lovelace/editor/config-elements/hui-conditional-card-editor.ts:327]: (unrelated, but we should add a border-radius)

**Review by bramkragten — APPROVED** [review]

**Comment by bramkragten** [comment]
We do need docs for this :-)

**Pull request merged**